File: src/entities/Episode.ts
import {
  Column,
  Entity,
  Index,
  JoinColumn,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
} from "typeorm";
import { Season } from "./Season";
import { Series } from "./Series";
import { SeriesHistory } from "./SeriesHistory";
import { SeriesServers } from "./SeriesServers";

@Index("pk_episode", ["episodeId"], { unique: true })
@Index(
  "episode_episode_number_season_id_series_id_key",
  ["episodeNumber", "seasonId", "seriesId"],
  { unique: true }
)
@Entity("episode", { schema: "dionysus" })
export class Episode {
  @PrimaryGeneratedColumn({ type: "bigint", name: "episode_id" })
  episodeId: string;

  @PrimaryGeneratedColumn({ type: "bigint", name: "series_id", unique: true })
  seriesId: string;

  @PrimaryGeneratedColumn({ type: "bigint", name: "season_id", unique: true })
  seasonId: string;

  @Column("integer", { name: "episode_number", unique: true })
  episodeNumber: number;

  @Column("character varying", {
    name: "episode_name",
    nullable: true,
    length: 100,
  })
  episodeName: string | null;

  @Column("character varying", {
    name: "episode_wallpaper",
    nullable: true,
    length: 480,
  })
  episodeWallpaper: string | null;

  @Column("character varying", { name: "summery", nullable: true, length: 480 })
  summery: string | null;

  @Column("integer", { name: "release_date", nullable: true })
  releaseDate: number | null;

  @Column("integer", { name: "episode_length", nullable: true })
  episodeLength: number | null;

  @ManyToOne(() => Season, (season) => season.episodes, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "season_id", referencedColumnName: "seasonId" }])
  season: Season;

  @ManyToOne(() => Series, (series) => series.episodes, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "series_id", referencedColumnName: "seriesId" }])
  series: Series;

  @OneToMany(() => SeriesHistory, (seriesHistory) => seriesHistory.episode)
  seriesHistories: SeriesHistory[];

  @OneToMany(() => SeriesServers, (seriesServers) => seriesServers.episode)
  seriesServers: SeriesServers[];
}

File: scrapper/scrapper.api/Entities/Episode.cs
using System;
using System.Collections.Generic;

namespace scrapper.api.Entities
{
    public partial class Episode
    {
        public Episode()
        {
            SeriesHistories = new HashSet<SeriesHistory>();
            SeriesServers = new HashSet<SeriesServer>();
        }

        public int EpisodeId { get; set; }
        public int Number { get; set; }
        public string? Name { get; set; }
        public string? Wallpaper { get; set; }
        public string? Summery { get; set; }
        public int? ReleaseDate { get; set; }
        public int? EpisodeLength { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int? SeasonId { get; set; }
        public int? SeriesId { get; set; }

        public virtual Season? Season { get; set; }
        public virtual Series? Series { get; set; }
        public virtual ICollection<SeriesHistory> SeriesHistories { get; set; }
        public virtual ICollection<SeriesServer> SeriesServers { get; set; }
    }
}
